Sony projector switch upgrade (#41468)
home-assistant/home-assistant.github.io: source/_integrations/sony_projector.markdown

---
title: Sony Projector
description: Instructions on how to integrate Sony projectors using SDCP as a media player in Home Assistant.
ha_category:
  - Multimedia
ha_iot_class: Local Polling
ha_config_flow: true
ha_release: 0.89
ha_domain: sony_projector
ha_platforms:
  - media_player
  - switch
ha_integration_type: integration
ha_quality_scale: legacy
---

> The Sony Projector integration allows Home Assistant to control the power state of SDCP-compatible, network-connected projectors from [Sony](https://www.sony.com/). It exposes the projector as a media player entity with on/off controls.

Reviewer: ```suggestion
The **Sony Projector** {% term integration %} allows Home Assistant to control the power state of SDCP-compatible, network-connected projectors from [Sony](https://www.sony.com/). It exposes the projector as a media player entity with on/off controls.
```

Author: Thanks for this; wasn't aware of the usage of that `term integration` piece!

I'll set a reminder to take a look for the rest of these to familiarize myself with them (unless you have a quick link handy)!

{% include integrations/config_flow.md %}

## Projector settings

> 

Reviewer: is this something I should do before adding the integration? Are these prerequisites?

Author: Yes, these are potential prerequisites for the projector to be discovered // usable by the integration.

Author: I've analyzed other integrations doco (mikrotik, tesla fleet, home_connect, etc) and moved the prerequisites before the config flow documentation.

Thanks!

Under your projector's web interface Advanced Menu, you may need to:
- Enable/start the PJ Talk service.
- Add the IP address of your Home Assistant server to the Host address field.

## Entities

- Media player: Represents the projector and supports turning the device on and off.
- Legacy switch (deprecated; transitional): If you still have a `switch:` YAML entry for `sony_projector`, a temporary compatibility switch entity is provided to keep existing automations working. It disappears automatically after you remove the YAML and restart. A repair notice appears only when your automations, scripts, scenes, or groups reference the legacy switch and will clear automatically after a reload or restart once no references remain.

## YAML configuration

> YAML configuration for `sony_projector` is no longer supported. If you previously configured this integration using YAML, remove any `sony_projector` entries from your `configuration.yaml` (including legacy `switch` or `media_player` sections) and set up the integration from Settings → Devices & Services.

Reviewer: ```suggestion
YAML configuration for `sony_projector` is no longer supported. If you previously configured this integration using YAML, remove any `sony_projector` entries from your `configuration.yaml` (including legacy `switch` or `media_player` sections) and set up the integration from {% my integrations title="**Settings** > **Devices & services**" %}.
```

After removing the YAML and restarting Home Assistant:

> - The temporary compatibility switch (if present) is removed automatically

Reviewer: ```suggestion
- The temporary compatibility switch (if present) is removed automatically.
```

> - The Repairs notice about the legacy switch clears automatically after a reload or restart once no automations or scripts reference it

Reviewer: ```suggestion
- The **Repairs** notice about the legacy switch clears automatically after a reload or restart once no automations or scripts reference it.
```

> Tip: to re-check immediately without restarting, reload the Sony Projector integration from Settings → Devices & Services.

Reviewer: ```suggestion
Tip: to re-check immediately without restarting, reload the Sony Projector integration from {% my integrations title="**Settings** > **Devices & services**" %}.
```

> To migrate existing automations, update them to call `media_player.turn_on` and `media_player.turn_off` on the projector entity. To find where the legacy switch is used, open Settings → Developer tools → Search and look up the switch entity to see related automations, scripts, scenes, and groups.

Reviewer: which tab in the developer tools?

Author: Thanks, I've clarified further in 4e98e51

---
title: Sony Projector
description: Instructions on how to integrate Sony projectors using SDCP as a media player in Home Assistant.
ha_category:
  - Multimedia
ha_iot_class: Local Polling
ha_config_flow: true
ha_release: 0.89
ha_domain: sony_projector
ha_platforms:
  - media_player
  - switch
ha_integration_type: integration
ha_quality_scale: legacy
---

The **Sony Projector** {% term integration %} allows Home Assistant to control the power state of SDCP-compatible, network-connected projectors from [Sony](https://www.sony.com/). It exposes the projector as a media player {% term entity %} with on/off controls.

## Prerequisites

> Under your projector's web interface Advanced Menu, you may need to:

Reviewer: ```suggestion
Under your projector's web interface **Advanced** menu, you may need to:
```
I don't know the application, but I assume Advanced is a UI string? We use bold to mark UI strings.

{% include integrations/config_flow.md %}

## Entities

- Media player: Represents the projector as an {% term entity %} and supports turning the device on and off.
- Legacy switch (deprecated; transitional): If you still have a `switch:` {% term YAML %} entry for `sony_projector`, a temporary compatibility switch {% term entity %} is provided to keep existing {% term automations %} working. It disappears automatically after you remove the {% term YAML %} and restart. A repair notice appears only when your {% term automations %}, {% term scripts %}, {% term scenes %}, or {% term groups %} reference the legacy switch and will clear automatically after a {% term reload %} or restart once no references remain.

## YAML configuration

{% term YAML %} configuration for `sony_projector` is no longer supported. If you previously configured this integration using {% term YAML %}, remove any `sony_projector` entries from your {% term "`configuration.yaml`" %} (including legacy `switch` or `media_player` sections) and set up the integration from {% my integrations title="**Settings** > **Devices & services**" %}.

> ### Resolving the Deprecation

Reviewer: ```suggestion
### Resolving the deprecation
```
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization), also in headings.

After removing the {% term YAML %} and restarting Home Assistant:
- The temporary compatibility switch (if present) is removed automatically.
- The **Repairs** notice about the legacy switch clears automatically after a {% term reload %} or restart once no automations or scripts reference it.

Tip: to re-check immediately without restarting, {% term reload %} the Sony Projector integration from {% my integrations title="**Settings** > **Devices & services**" %}.

### Migrating existing associations to new component

To migrate existing {% term automations %}, update them to call `media_player.turn_on` and `media_player.turn_off` on the projector {% term entity %}. To find where the legacy switch is used:

> - Open Developer tools → States → Filter Entities.

Reviewer: ```suggestion
- Open {% my developer_states title="**Developer Tools** > **States**" %} and find the **Filter entities** filter text field.
```

> - Look up the switch {% term entity %} via the Entity Column Filter (ie. `switch.*projector`).

Reviewer: ```suggestion
- Look up the switch {% term entity %} via the **Filter entities** filter text field (for example, `switch.*projector`).
```

> - Click the (i) to open the entity details pane.
> - Click the menu button → `Related` see related {% term automations %}, {% term scripts %}, {% term scenes %}, and {% term groups %}.

Reviewer: ```suggestion
- To open the entity details pane, select the {% icon "mdi:information-outline" %}.
- To see related {% term automations %}, {% term scripts %}, {% term scenes %}, and {% term groups %}, select the three dots {% icon "mdi:dots-vertical" %} menu, then select **Related**.
```
